### PR TITLE
Mapi 1022 update notification client

### PR DESCRIPTION
**Pull request by alexduf**
- Move the generation of the legacy notification to the client
 - Update sbt
 - update notification client
 - force timeouts to be bigger

### CODE REVIEW

File: sbt-tc [hunk 1]
```diff
@@ -14,5 +14,5 @@ cat /dev/null | java -Xmx1G -XX:MaxPermSize=250m -XX:+UseCompressedOops \
     -Dsbt.boot.directory=$SBT_BOOT_DIR \
     $PROXY_CONF \
     -Dfile.encoding=UTF8 \
-    -jar sbt-launch-0.*.jar "$@"
+    -jar sbt-launch*.jar "$@"
```
Comment (agulowaty) [inline sbt-tc]: * is not needed

File: src/main/scala/com/gu/mobile/notifications/content/ExpirationRequest.scala [hunk 2]
```diff
@@ -1,11 +1,30 @@
 package com.gu.mobile.notifications.content
 
-import com.gu.mobile.notifications.client.models.Topic
+import com.gu.mobile.notifications.client.models.{TopicTypes, TopicType, Topic}
+import com.gu.mobile.notifications.client.models.TopicTypes._
 import spray.json._
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/content/ExpirationRequest.scala]: shouldn't we migrate to play.json ?

File: src/main/scala/com/gu/mobile/notifications/content/SNSQueue.scala [hunk 3]
```diff
@@ -16,7 +17,7 @@ trait SNSQueue extends Logging {
 
   lazy val topic = MobileNotificationsConfig.snsTopic
 
-  def publishNotifications(notification: Notification) {
+  def publishNotifications(notification: NotificationPayload) {
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/content/SNSQueue.scala]: please rename argument to `payload`

File: src/main/scala/com/gu/mobile/notifications/content/actors/MessageSendingActor.scala [hunk 4]
```diff
@@ -3,12 +3,12 @@ package com.gu.mobile.notifications.content.actors
...
 import scala.util.{Failure, Success}
 import com.gu.mobile.notifications.client.ApiClient
-import com.gu.mobile.notifications.content.lib.NotificationBuilder
+import com.gu.mobile.notifications.content.lib.ContentAlertPayloadBuilder
 import com.gu.contentapi.client.model.Content
 import akka.actor.{Actor, Props}
 import com.gu.mobile.notifications.content.management.Metrics
 import com.gu.util.liveblogs.Block
-import com.gu.mobile.notifications.client.models.Notification
+import com.gu.mobile.notifications.client.models.{ContentAlertPayload}
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/content/actors/MessageSendingActor.scala]: moustaches not needed

File: src/main/scala/com/gu/mobile/notifications/content/actors/MessageSendingActor.scala [hunk 5]
```diff
@@ -24,11 +24,11 @@ class MessageSendingActor(notificationsClient: ApiClient) extends Actor with Log
...
 
   def receive = {
-    case Message(item, None) => sendNotification(NotificationBuilder(item))
-    case Message(item, Some(block)) => sendNotification(NotificationBuilder(item, block))
+    case Message(item, None) => sendNotification(ContentAlertPayloadBuilder(item))
+    case Message(item, Some(block)) => sendNotification(ContentAlertPayloadBuilder(item, block))
   }
 
-  private def sendNotification(notification: Option[Notification]) {
+  private def sendNotification(notification: Option[ContentAlertPayload]) {
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/content/actors/MessageSendingActor.scala]: please rename argument

File: src/main/scala/com/gu/mobile/notifications/content/lib/ContentAlertPayloadBuilder.scala [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  private def getTopicType(tagType: String): Option[TopicType] = tagType match {
+    case "contributor" => Some(TagContributor)
+    case "keyword" => Some(TagKeyword)
+    case "series" => Some(TagSeries)
+    case "blog" => Some(TagBlog)
+    case _ => None
+  }
+
+  private def tagToTopic(tag: Tag): Option[Topic] = {
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/content/lib/ContentAlertPayloadBuilder.scala]: outer braces not needed

File: src/main/scala/com/gu/mobile/notifications/content/ExpirationRequest.scala (hunk 2 above)
Comment (alexduf) [inline src/main/scala/com/gu/mobile/notifications/content/ExpirationRequest.scala]: That would be great but it's out of scope

File: src/main/scala/com/gu/mobile/notifications/content/SNSQueue.scala (hunk 3 above)
Comment (alexduf) [inline src/main/scala/com/gu/mobile/notifications/content/SNSQueue.scala]: Well we are still publishing a notification

File: src/main/scala/com/gu/mobile/notifications/content/actors/MessageSendingActor.scala (hunk 5 above)
Comment (alexduf) [inline src/main/scala/com/gu/mobile/notifications/content/actors/MessageSendingActor.scala]: Same here, ContentAlertPayload extends Notification trait

**Comment by DiegoVazquezNanini** [comment]
Nice refactor. It makes the code simpler and cleaner. 
I don't really understand if it's functionally equivalent to the previous code because the test are also changing. :), but anyway nice work @pvighi and @alexduf
:+1:

**Pull request merged**